rpc/src/lib.rs

```diff
@@ -420,8 +420,20 @@ impl<B, C, SC, P, CT, BE> EthApiT for EthApi<B, C, SC, P, CT, BE> where
 		unimplemented!("logs");
 	}
 
-	fn work(&self, _: Option<u64>) -> Result<Work> {
-		unimplemented!("work");
+	fn work(&self) -> Result<Work> {
+		let header = self
+			.select_chain
+			.best_chain()
+			.map_err(|_| internal_err("fetch header failed"))?;
+		
+		Ok(Work {
+			pow_hash: H256::from(header.hash()),
```

Comment:
Technically I think this should be the Ethereum header hash, following convention of the rest of the codebase. This can also be `H256::default()` given that getting work in a non-PoW chain is basically useless.